### PR TITLE
Update signature of compare/diff built-in functions step by step

### PR DESCRIPTION
**Pull request by k1LoW**
ref: https://github.com/k1LoW/runn/issues/1047

Because variadic arguments like Go cannot be used in expr-lang.

While accepting both arguments, deprecate the variadic argument type.

### CODE REVIEW

**Comment by k1LoW** [comment]
@k2tzumi
I'd be glad to get your feedback on this change.

File: builtin/diff.go [hunk 1]
```diff
@@ -10,9 +10,36 @@ import (
...
 	"github.com/itchyny/gojq"
+	"github.com/k1LoW/runn/internal/deprecation"
 )
 
-func Diff(x, y any, ignorePaths ...string) (string, error) {
+func Diff(x, y any, ignores ...any) (string, error) {
+	var ignoreSpecifiers []string
+	if len(ignores) > 1 {
+		deprecation.AddWarning("diff/compare", "diff/compare(x, y, ignores...string) is deprecated. Use diff/compare(x, y, ignores []string) instead.")
+	}
```
Comment (k2tzumi) [inline builtin/diff.go:20]: I understood that it would be phased out.

File: builtin/diff.go [hunk 2]
```diff
@@ -10,9 +10,36 @@ import (
...
+			deprecation.AddWarning("diff/compare", "diff/compare(x, y, ignores...string) is deprecated. Use diff/compare(x, y, ignores []string) instead.")
+			ignoreSpecifiers = append(ignoreSpecifiers, v)
+		case []any:
+			for _, vv := range v {
+				s, ok := vv.(string)
+				if !ok {
+					return "", fmt.Errorf("invalid ignore specifiers: %v", vv)
+				}
+				ignoreSpecifiers = append(ignoreSpecifiers, s)
+			}
```
Comment (k2tzumi) [inline builtin/diff.go:33]: Currently accepting ANY for compatibility, so it needs to be checked.

File: builtin/diff.go [hunk 3]
```diff
@@ -25,7 +52,7 @@ func Diff(x, y any, ignorePaths ...string) (string, error) {
 		return "", err
 	}
 
-	jqIgnorePaths, cmpIgnoreKeys := impl.splitIgnoreSpecifiers(ignorePaths)
+	jqIgnorePaths, cmpIgnoreKeys := impl.splitIgnoreSpecifiers(ignoreSpecifiers)
```
Comment (k2tzumi) [inline builtin/diff.go:55]: This process will also be unnecessary in the future.

File: testdata/book/builtin_compare.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+desc: For compare() built-in function
+steps:
+  merge:
+    test: |
+      compare(
+        {"a": 1, "b": 3, "c": 5},
+        {"a": 1, "b": 2, "c": 4},
+        "b", "c"
+      )
```
Comment (k2tzumi) [inline testdata/book/builtin_compare.yml:9]: This test here will fail in the future.

**Pull request merged**